src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,204 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A Domain for testing purposes.
+ *
+ * \image html binary_compact_object_domain.png "A BHNS domain."
+ *
+ * Creates a 3D Domain consisting of four/five layers of blocks:
+ * - 0: The block at the center of the compact object, if not excised.
+ * - 1: The first layer, has the blocks for the compact object.
+ * - 2: The second layer, has the blocks that surround the object with a cube.
+ * - 3: The third layer, has the blocks that surround each cube with a
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
+ *
+ * `ObjectA` and `ObjectB` refer to the Shell or Sphere making up each compact
+ * object.
+ * `enveloping cube` and `enveloping sphere` refer to the bounding surfaces of
+ * the outer Sphere of ten wedges making up the last layer of the domain.
+ *
+ * The plane along which the domain divides into two hemispheres is referred
+ * to as the "cutting plane".
+ */
+
+template <typename TargetFrame>
+class BinaryCompactObject : public DomainCreator<3, TargetFrame> {
+ public:
+  struct InnerRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Coordinate radius of this compact object."};
+  };
+
+  struct OuterRadiusObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "Coordinate radius of this compact object."};
+  };
+
+  struct XCoordObjectA {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of this compact object."};
+  };
+
+  struct ExciseInteriorA {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
+  };
+
+  struct InnerRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Areal radius of this compact object."};
+  };
+
+  struct OuterRadiusObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "Areal radius of this compact object."};
+  };
+
+  struct XCoordObjectB {
+    using type = double;
+    static constexpr OptionString help = {
+        "x-coordinate of this compact object."};
+  };
+
+  struct ExciseInteriorB {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Set to `true` for a BH, `false` otherwise."};
```

Comment:
Help-text should explain what the option does.  Here, it only says how one should set it.  Perhaps "If 'true', layers 0 and 1 will be absent.  Set to `true` for a BH, ...."